src/Domain/Command/CheckInUser.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Building\Domain\Command;
+
+use Prooph\Common\Messaging\Command;
+use Rhumsaa\Uuid\Uuid;
+
+final class CheckInUser extends Command
+{
+    /** @var string */
+    private $username;
+
+    /** @var Uuid */
+    private $buildingId;
+
+    private function __construct(Uuid $buildingId, string $username)
+    {
+        $this->init();
+
+        $this->buildingId = $buildingId;
+        $this->username   = $username;
+    }
+
+    public static function intoBuilding(Uuid $buildingId, string $username) : self
```

Comment:
A named constructor always helps in keeping the parameters and behavior well described, hence it is preferable over `__construct`.

Also, `__construct` can only be declared once, which makes it a very hard BC boundary for downstream consumers, if it is ever altered due to object internal state requirements changing.